api/client/registry/login.go

```diff
@@ -52,7 +53,16 @@ func runLogin(dockerCli *client.DockerCli, opts loginOptions) error {
 
 	var serverAddress string
 	var isDefaultRegistry bool
-	if opts.serverAddress != "" {
+
+	// when a user supplied registry url is the default docker.io registry just ignore it
+	// without this check the login will succed, but the credentials will not be saved properly and can't be used
+	// handle urls without schema for backward compatibility and pass the integration tests
+	u, _ := url.Parse(opts.serverAddress)
+	if u.Host == "" {
+		u, _ = url.Parse("http://" + opts.serverAddress)
```

Comment:
The `http://` is only added here to form a valid URL. Docker doesn't store the `http(s)://`, and defaults to using https:// at all times, _unless_ a registry is added to "insecure registries", in which case it allows falling back to non-https connections